File: app/src/main/java/com/funny/translation/BaseActivity.java
package com.funny.translation;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity
{

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		// TODO: Implement this method
		super.onCreate(savedInstanceState);
		//setTheme(R.style.AppTheme_NoActionBar);
	}

	public void moveToActivity(Class activity){
		Intent intent=new Intent();
		intent.setClass(this,activity);
		startActivity(intent);
	}

	public void moveToActivityForResult(Class activity,Intent intent,int startCode){
		intent.setClass(this,activity);
		startActivityForResult(intent,startCode);
	}
	
}

File: base-core/src/main/java/com/funny/translation/BaseActivity.kt
package com.funny.translation

import android.content.Context
import android.os.Bundle
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.AppCompatActivity
import com.funny.translation.helper.LocaleUtils
import com.smarx.notchlib.NotchScreenManager

open class BaseActivity : AppCompatActivity() {
    private lateinit var callback: OnBackPressedCallback

    companion object {
        private const val TAG = "BaseActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 状态栏沉浸
        NotchScreenManager.getInstance().setDisplayInNotch(this)

        callback = object : OnBackPressedCallback(true) {
            override fun handleOnBackPressed() {
                finish()
            }
        }
        onBackPressedDispatcher.addCallback(this, callback)

    }

    override fun attachBaseContext(newBase: Context?) {
        val context = newBase?.let {
            LocaleUtils.getWarpedContext(it)
        }
        super.attachBaseContext(context)
    }
}